File: app/src/main/java/com/jsqix/dongqing/gank/fragment/WebFragment.java
package com.jsqix.dongqing.gank.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.jsqix.dongqing.gank.R;

/**
 * Created by dongqing on 2017/1/10.
 */

public class WebFragment extends BaseFragment {
    private WebView webview;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_web, container, false);
        return super.onCreateView(inflater, container, savedInstanceState);
    }

    @Override
    protected void initView() {
        webview = (WebView) view.findViewById(R.id.webview);
        WebSettings settings = webview.getSettings();
        settings.setJavaScriptEnabled(true);
        settings.setUseWideViewPort(true);
        webview.loadUrl("http://gank.io/xiandu");
        webview.setWebViewClient(new MyWebView());
    }

    class MyWebView extends WebViewClient {

        @Override
        public boolean shouldOverrideUrlLoading(WebView view, String url) {
            view.loadUrl(url);
            return super.shouldOverrideUrlLoading(view, url);
        }
    }

}

File: app/src/main/java/com/dongqing/gank/kotlin/ui/fragment/WebFragment.kt
package com.dongqing.gank.kotlin.ui.fragment


import android.webkit.WebView
import android.webkit.WebViewClient
import com.dongqing.gank.kotlin.R
import com.dongqing.gank.kotlin.ui.view.IFloatView
import kotlinx.android.synthetic.main.fragment_web.*

class WebFragment : BaseFragment(), IFloatView {
    override fun fastMoveToTop() {
        ns_view.smoothScrollTo(0, 0)
    }

    override fun initView() {
        val settings = webview.settings
        settings.javaScriptEnabled = true
        settings.useWideViewPort = true
        settings.userAgentString = "User-Agent: Mozilla/5.0 (Linux; Android 7.0; ZUK Z2131 Build/NRD90M; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/53.0.2785.49 Mobile MQQBrowser/6.2 TBS/043015 Safari/537.36 MicroMessenger/6.5.3.980 NetType/WIFI Language/zh_CN"
        webview.loadUrl("http://gank.io/xiandu")
        webview.webViewClient = MyWebView()
    }

    override fun getLayoutId(): Int {
        return R.layout.fragment_web
    }

    internal inner class MyWebView : WebViewClient() {

        override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
            view.loadUrl(url)
            return super.shouldOverrideUrlLoading(view, url)
        }
    }
}
